### PR TITLE
Add path filter to joomla updates download process for the temp folder

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

The JSST has been contacted about an missing path filter whithin the joomla download process. Given that an successfull attack requires a Super User access to change the tmp path setting and also to trigger the update itself the JSST decided to move this patch to the public tracker.

### Testing Instructions

- apply the patch
- Try to install an update or to "reinstall the core files"
- Make sure the update works as expected
- try to install an update via this custom update server (https://ci.joomla.org/artifacts/joomla/joomla-cms/staging/32076/downloads/39782/pr_list.xml) 

### Actual result BEFORE applying this Pull Request

The upgrade works as expected

### Expected result AFTER applying this Pull Request

The upgrade works as expected

### Documentation Changes Required

none

### CODE REVIEW

**Comment by nibra** [comment]
You can then replace the whole switch statement with
`$result = parent::clean($source, $type);`

**Comment by zero-24** [comment]
Right even better 👍

File: administrator/components/com_joomlaupdate/models/default.php [hunk 1]
```diff
@@ -301,7 +301,7 @@ public function download()
 
 		// Find the path to the temp directory and the local package.
 		$config   = JFactory::getConfig();
-		$tempdir  = $config->get('tmp_path');
+		$tempdir  = (string) InputFilter::getInstance([], [], 1, 1)->clean($config->get('tmp_path'), 'path');
```
Comment (zero-24) [inline administrator/components/com_joomlaupdate/models/default.php]:
```suggestion
		$tempdir  = (string) InputFilter::getInstance(array(), array(), 1, 1)->clean($config->get('tmp_path'), 'path');
```

**Comment by richard67** [comment]
@zero-24 Shall we test this, or shall we wait until the Todo's are done? If test: Could you add the missing link to an update package in the testing instructions (currently "(to be generated)")?

**Comment by zero-24** [comment]
This has to wait for the archive package to be merged and the path filter patched as noted above. The package i mean is the one generated by drone.

File: administrator/components/com_joomlaupdate/models/default.php [hunk 2]
```diff
@@ -301,7 +301,7 @@ public function download()
 
 		// Find the path to the temp directory and the local package.
 		$config   = JFactory::getConfig();
-		$tempdir  = $config->get('tmp_path');
+		$tempdir  = (string) InputFilter::getInstance(array(), array(), 1, 1)->clean($config->get('tmp_path'), 'path');
```
Comment (richard67) [inline administrator/components/com_joomlaupdate/models/default.php:306]: Either we need to change to `JFilterInput` here, or we need to add a `use Joomla\CMS\Filter\InputFilter;` at the top.

Comment (richard67) [reply administrator/components/com_joomlaupdate/models/default.php:306]: Otherwise we get a `Class 'InputFilter' not found` error.

Comment (zero-24) [reply administrator/components/com_joomlaupdate/models/default.php:306]: Done: https://github.com/joomla/joomla-cms/pull/32076/commits/6bc09ac5a3de6863f23440d4079b3c16bd3e2ceb

**Comment by richard67** [comment]
@zero-24 I've updated in the description the link to the update package for this PR so it points to the latest build.

**Comment by richard67** [comment]
Test on Linux was ok, but on Windows it was failing. See https://github.com/joomla-framework/filter/pull/40 for the fix in the framework package.

**Comment by richard67** [comment]
Now we have to wait for #32206 to be merged.

**Comment by zero-24** [comment]
I have just merged the filter package and updated the branch here so this is ready to be tested again.

**Comment by richard67** [comment]
I've updated in the description the link to the update package for this PR so it points to the latest build.

**Comment by zero-24** [comment]
> I've updated in the description the link to the update package for this PR so it points to the latest build.

Thanks was about to do that too as we have to wait for it to be generated👍

**Comment by richard67** [comment]
I had to do it again because I had to restart drone.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on adab40bad2721f8023ff7e1df1ad82c3d733f9e1<br /><br />Tested with 2 server environments, one Linux with PHP 7.3, and one Windows with PHP 7.4.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32076">issues.joomla.org/tracker/joomla-cms/32076</a>.</sub>

**Comment by richard67** [comment]
@zero-24 I've removed the "Todo's" section from the description because that stuff has been done meanwhile.

File: administrator/components/com_joomlaupdate/models/default.php [hunk 3]
```diff
@@ -12,6 +12,8 @@
 jimport('joomla.filesystem.folder');
 jimport('joomla.filesystem.file');
 
+use Joomla\CMS\Filter\InputFilter;
```
Comment (Quy) [inline administrator/components/com_joomlaupdate/models/default.php]: Move after line 10?

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on adab40bad2721f8023ff7e1df1ad82c3d733f9e1<br /><br />Tested on win10, with php8, following the testing instructions<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32076">issues.joomla.org/tracker/joomla-cms/32076</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32076">issues.joomla.org/tracker/joomla-cms/32076</a>.</sub>

**Comment by richard67** [comment]
@zero-24 Could you check @Quy 's suggestion above and fix it? https://github.com/joomla/joomla-cms/pull/32076#discussion_r568721616 . It will not change RTC status.

File: administrator/components/com_joomlaupdate/models/default.php (hunk 3 above)
Comment (zero-24) [reply administrator/components/com_joomlaupdate/models/default.php]: Moved https://github.com/joomla/joomla-cms/pull/32076/commits/ce05326203a1f2a759c1fe16e1dcd5066ff6b8dc

**Comment by zero-24** [comment]
> @zero-24 Could you check @Quy 's suggestion above and fix it? [#32076 (comment)](https://github.com/joomla/joomla-cms/pull/32076#discussion_r568721616) . It will not change RTC status.

Moved with: https://github.com/joomla/joomla-cms/pull/32076/commits/ce05326203a1f2a759c1fe16e1dcd5066ff6b8dc

**Comment by richard67** [comment]
Previous tests are still valid since last change after tests was code style only. I've restored the test results in the tracker.

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**